src/libs/auth.ts

```diff
@@ -97,71 +127,206 @@ const revokeTokens = async () => {
+          ...labelOldAuthToken(oldAuthTokenMetadata),
+          refreshTokenId: authStore.get().refreshTokenMetadata.id,
+          refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+          refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+        });
+      },
+    ],
+    [
+      'error',
+      () => {
+        Ajax().Metrics.captureEvent(Events.user.authTokenLoad.error, {
+          // we could potentially log the reason, but I don't know if that data is safe to log
+          ...labelOldAuthToken(oldAuthTokenMetadata),
+          refreshTokenId: authStore.get().refreshTokenMetadata.id,
+          refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+          refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+        });
+      },
+    ]
+  );
+
+  return loadedAuthTokenState;
+};
+
+const tryLoadAuthToken = async (includeBillingScope = false, popUp = false): Promise<AuthTokenState> => {
+  const args: ExtraSigninRequestArgs = getSigninArgs(includeBillingScope);
+  const authInstance = getAuthInstance();
+  const loadedAuthTokenResponse: User | null | AuthTokenErrorState = await (popUp
+    ? // returns Promise<User | null>, attempts to use the refresh token to get a new authToken
+      authInstance.signinPopup(args)
+    : authInstance.signinSilent(args)
+  ).catch((e) => {
+    const errorMsg = 'An unexpected exception occurred while attempting to load auth credentials.';
+    console.error(errorMsg);
+    console.error(e);
+    const errorState: AuthTokenErrorState = {
+      errorMsg,
+      reason: e,
+      status: 'error',
+    };
+    return errorState;
   });
-  return user;
+  if (loadedAuthTokenResponse instanceof User) {
+    return {
+      status: 'success',
+      user: loadedAuthTokenResponse,
+    };
+  }
+  if (loadedAuthTokenResponse === null) {
+    return {
+      status: 'expired',
+    };
+  }
+
+  return loadedAuthTokenResponse;
 };
 
-export const reloadAuthToken = (includeBillingScope = false) => {
-  const args = getSigninArgs(includeBillingScope);
-  const tokenMetadata = authStore.get().authTokenMetadata;
-  let authReloadSuccessful = true;
-  return getAuthInstance()
-    .signinSilent(args)
-    .catch((e) => {
-      console.error('An unexpected exception occurred while attempting to refresh auth credentials.');
-      console.error(e);
-      authReloadSuccessful = false;
-      return false;
-    })
-    .finally(() => {
-      Ajax().Metrics.captureEvent(Events.userAuthTokenReload, {
-        authReloadSuccessful,
-        oldAuthTokenCreatedAt: Utils.formatTimestampInSeconds(tokenMetadata.createdAt),
-        oldAuthTokenExpiresAt: Utils.formatTimestampInSeconds(tokenMetadata.expiresAt),
-      });
-    });
+const labelOldAuthToken = (oldAuthTokenMetadata: TokenMetadata) => {
```

Comment:
Nit on naming: `labelX` makes me think this is modifying `X`.

Suggest something that more clearly suggests this is returning the labels vs doing the labelling. `getOldAuthTokenLabels`, etc.

Same comment for `labelTimestampMetric`.